src/WebApi/Controllers/CinemasController.cs

```diff
@@ -1,50 +1,87 @@
 
 namespace WebApi.Controllers
 {
     [Route("api/[controller]")]
     public class CinemasController : Controller
     {
+        [NotNull]
+        private readonly ICinemasService _cinemasService;
+
+        public CinemasController([NotNull] ICinemasService cinemasService)
+        {
+            _cinemasService = cinemasService;
+        }
+
         // GET /cinemas
         [HttpGet]
-        public IEnumerable<CinemaModelResponse> Get()
+        public async Task<IActionResult> Get()
         {
-            //User.FindFirst(ClaimTypes.NameIdentifier);
-            List<CinemaModelResponse> cinemas = new List<CinemaModelResponse>();
-            return cinemas;
+            IEnumerable<BlCinemaModelResponse> cinemas = await _cinemasService.GetCinemas();
+
+            return Ok(
+                cinemas.Select(Mapper.Map<ApiCinemaModelResponse>)
+                );
         }
 
         // GET /cinemas/{id}
         [HttpGet("{id:int}")]
-        public CinemaModelResponse Get(int id)
+        public async Task<IActionResult> Get(int id)
         {
-            var cinema = new CinemaModelResponse();
-            return cinema;
+            BlCinemaModelResponse cinema = await _cinemasService.GetCinemaById(id);
+
+            if (cinema == null)
+               return BadRequest();
+
+            return Ok(
+                Mapper.Map<ApiCinemaModelResponse>(cinema)
+                );
         }
 
         // GET /cinemas/{id}/halls
-        [HttpGet]
-        public IEnumerable<HallModel> GetHalls(){
-            List<HallModel> halls = new List<HallModel>();
-            return halls;
+        [HttpGet("{id:int}/halls")]
+        public async Task<IActionResult> GetHalls(int id)
+        {
+            IEnumerable<BlHallModelResponse> halls = await _cinemasService.GetHalls(id);
+
+            List<ApiHallModelResponse> results =
+                halls.Select(hall => new ApiHallModelResponse(
+                    hall.Id,
+                    hall.CinemaId,
+                    hall.Name,
+                    hall.Places.Select(Mapper.Map<PlaceModelResponseForHall>).ToArray(),
+                    hall.HallSchemeModels.Select(Mapper.Map<ApiHallSchemeModelResponse>).ToArray())
+                ).ToList();
+
+            return Ok(results);
         }
 
         // PUT /cinemas
         [HttpPut]
-        public IActionResult Put([FromBody]CinemaModelRequest cinema)
+        public async Task<IActionResult> Put([FromBody]ApiCinemaModelRequest cinema)
         {
-            return StatusCode((int)HttpStatusCode.Created);
-        }
+            BlCinemaModelRequest cinemaRequest = new BlCinemaModelRequest(cinema.Name, cinema.City);
 
-        // DELETE /cinemas/{id}
-        [HttpDelete]
-        public IActionResult Delete(int id)
-        {
-            return StatusCode((int)HttpStatusCode.Accepted);
+            BlCinemaModelResponse cinemaResponse = await _cinemasService.AddCinema(cinemaRequest);
```

Comment:
Put should be idempotent. AddCinema according to its name is not idempotent. Should be used something like AddOrUpdateCinema